gitlab/client.py

```diff
@@ -832,15 +835,33 @@ def http_list(
         """
         query_data = query_data or {}
 
-        # In case we want to change the default behavior at some point
-        as_list = True if as_list is None else as_list
+        # Don't allow both `as_list` and `iterator` to be set.
+        if as_list is not None and iterator is not None:
+            raise gitlab.exceptions.GitlabError(
+                error_message=(
+                    "`as_list` and `iterator` are mutually exclusive. Only one "
+                    "argument can be set. Recommend to only set `iterator` argument"
+                )
+            )
+
+        if as_list is not None:
+            iterator = not as_list
+            utils.warn(
+                message=(
+                    f"`as_list={as_list}` has been deprecated. Please use "
+                    f"`iterator={iterator}` instead"
```

Comment:
Similar here, we have this:
https://github.com/python-gitlab/python-gitlab/blob/8867ee59884ae81d6457ad6e561a0573017cf6b2/gitlab/v4/objects/projects.py#L555-L573

```suggestion
                    f"`as_list={as_list}` is deprecated and will be removed in a future version.
                    f"Use `iterator={iterator}` instead."
```

Reply:
Done